### ISSUE TITLE
Local development bug: Pre-commit install error

### ISSUE THREAD

**Issue by machikoyasuda** [cal-itp/eligibility-server#83]
I am unable to get `pre-commit` running in my local devcontainer. This is the error log I get:

### version information

```
pre-commit version: 2.19.0
git --version: git version 2.30.2
sys.version:
    3.9.13 (main, May 18 2022, 04:17:41) 
    [GCC 10.2.1 20210110]
sys.executable: /usr/local/bin/python
os.name: posix
sys.platform: linux
```

### error information

```
An unexpected error has occurred: CalledProcessError: command: ('/root/.cache/pre-commit/repofu2cxgzn/py_env-python3.9/bin/python', '-mpip', 'install', '.')
return code: 1
expected return code: 0
stdout:
    Processing /root/.cache/pre-commit/repofu2cxgzn
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Collecting ruamel.yaml>=0.15
      Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
    Collecting toml
      Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
    Collecting ruamel.yaml.clib>=0.2.6
      Using cached ruamel.yaml.clib-0.2.6.tar.gz (180 kB)
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'error'
    
stderr:
      error: subprocess-exited-with-error
      
      × python setup.py egg_info did not run successfully.
      │ exit code: 1
      ╰─> [3 lines of output]
          sys.argv ['/tmp/pip-install-jxph4ctq/ruamel-yaml-clib_c3f62e16a11e4867877ec98577ee7f67/setup.py', 'egg_info', '--egg-base', '/tmp/pip-pip-egg-info-bdkc5a4p']
          test compiling /tmp/tmp_ruamel_n_ndon63/test_ruamel_yaml.c -> test_ruamel_yaml compile error: /tmp/tmp_ruamel_n_ndon63/test_ruamel_yaml.c
          Exception: command 'gcc' failed: No such file or directory
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: metadata-generation-failed
    
    × Encountered error while generating package metadata.
    ╰─> See above for output.
    
    note: This is an issue with the package mentioned above, not pip.
    hint: See above for details.
    
```

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pre_commit/error_handler.py", line 73, in error_handler
    yield
  File "/usr/local/lib/python3.9/site-packages/pre_commit/main.py", line 389, in main
    return run(args.config, store, args)
  File "/usr/local/lib/python3.9/site-packages/pre_commit/commands/run.py", line 414, in run
    install_hook_envs(to_install, store)
  File "/usr/local/lib/python3.9/site-packages/pre_commit/repository.py", line 223, in install_hook_envs
    _hook_install(hook)
  File "/usr/local/lib/python3.9/site-packages/pre_commit/repository.py", line 79, in _hook_install
    lang.install_environment(
  File "/usr/local/lib/python3.9/site-packages/pre_commit/languages/python.py", line 221, in install_environment
    helpers.run_setup_cmd(prefix, install_cmd)
  File "/usr/local/lib/python3.9/site-packages/pre_commit/languages/helpers.py", line 51, in run_setup_cmd
    cmd_output_b(*cmd, cwd=prefix.prefix_dir, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/pre_commit/util.py", line 146, in cmd_output_b
    raise CalledProcessError(returncode, cmd, retcode, stdout_b, stderr_b)
pre_commit.util.CalledProcessError: command: ('/root/.cache/pre-commit/repofu2cxgzn/py_env-python3.9/bin/python', '-mpip', 'install', '.')
return code: 1
expected return code: 0
stdout:
    Processing /root/.cache/pre-commit/repofu2cxgzn
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Collecting ruamel.yaml>=0.15
      Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
    Collecting toml
      Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
    Collecting ruamel.yaml.clib>=0.2.6
      Using cached ruamel.yaml.clib-0.2.6.tar.gz (180 kB)
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'error'
    
stderr:
      error: subprocess-exited-with-error
      
      × python setup.py egg_info did not run successfully.
      │ exit code: 1
      ╰─> [3 lines of output]
          sys.argv ['/tmp/pip-install-jxph4ctq/ruamel-yaml-clib_c3f62e16a11e4867877ec98577ee7f67/setup.py', 'egg_info', '--egg-base', '/tmp/pip-pip-egg-info-bdkc5a4p']
          test compiling /tmp/tmp_ruamel_n_ndon63/test_ruamel_yaml.c -> test_ruamel_yaml compile error: /tmp/tmp_ruamel_n_ndon63/test_ruamel_yaml.c
          Exception: command 'gcc' failed: No such file or directory
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: metadata-generation-failed
    
    × Encountered error while generating package metadata.
    ╰─> See above for output.
    
    note: This is an issue with the package mentioned above, not pip.
    hint: See above for details.
```

**Comment by afeld** [comment]
Try adding `build-essential` to

https://github.com/cal-itp/eligibility-server/blob/547d2f3cc6ad3e7c370afe1387be792c9f51cfab/Dockerfile.dev#L8

**Comment by thekaveman** [comment]
This seems like something to look further into:

```
command: ('/root/.cache/pre-commit/repofu2cxgzn/py_env-python3.9/bin/python', '-mpip', 'install', '.')
```

`pre-commit` installs a script file at `.git/hooks/pre-commit` that I think is the one running the above command.

* Maybe try deleting and previous `pre-commit` script in your git repo: `rm -rf .git/hooks/pre-commit`
* When is the last time you built the underlying Docker container locally? `docker compose build server`
* Same for the devcontainer `docker compose build dev`
* Maybe try deleting the `vscode` volume that VS Code creates in Docker for devcontainer purposes. Restart from scratch essentially?

I would be surprised if this was due to a missing/needed package in the devcontainer. If that were the case, I would expect you to get the same error building/running the `benefits` devcontainer.

**Comment by machikoyasuda** [comment]
@thekaveman To find out for sure, I ran two tests:

0. I rebuilt a new devcontainer and server, with the current Dockerfile.dev, to confirm that I'm unable to get pre-commit running.
1. Then, I built a new devcontainer with the same exact build step as benefit's: https://github.com/cal-itp/benefits/blob/b070bd0ad2f5e63d19085a808d414e3dec3676e7/.devcontainer/Dockerfile#L10

Installs Node and Cypress dependencies:
```
RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash

RUN apt-get update \
    && apt-get install -qq nodejs npm libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb curl git jq ssh
```

And started up the devcontainer. This time, I was able to get pre-commit running.

2. Then, I ran this PR: https://github.com/cal-itp/eligibility-server/pull/85's and recreated this devcontainer, where I was able to get pre-commit running.

```
RUN apt-get update \
    && apt-get install -qq --no-install-recommends curl git jq ssh build-essential \
```

Since just adding `build-essential` is simpler than adding all the Node deps, I'm going with this approach.
